where-to-travel/src/main/webapp/script.js

```diff
@@ -376,75 +383,134 @@ function populatePlaces(placeArray) {
-    // especially for adding elements.
-    let cardElement = $(htmlContent).click(function(event) {
-      if(event.target.nodeName != 'SPAN') {
-        toggleFocusOff();
-        selectLocationMarker($(this).attr('placeName'));
-        $(this).addClass('active-card');
-       focusedCard = this;
+     // Check to see if it is a saved place, if so make the star pressed 
+      if(savedPlacesSet.has(place.place_id) === true) {
+        cardElement.find('.icon').addClass('press');
       }
-    });
-    $('#' + SCROLL_ID).append(cardElement);
-
-    // Add events to focus card and pin
-    placeMarker.addListener('click', function () {
-      toggleFocusOff();
-      focusedPin = placeMarker;
-      selectLocationCard(placeMarker.getTitle());
-      placeMarker.setIcon(SELECTED_PIN_PATH);
-      focusedCard.scrollIntoView({behavior: 'smooth', block: 'center'});
-    });
+      // Add events to focus card and pin
+      placeMarker.addListener('click', function () {
+        toggleFocusOff();
+        focusedPin = placeMarker;
+        selectLocationCard(placeMarker.getTitle());
+        placeMarker.setIcon(SELECTED_PIN_PATH);
+        focusedCard.scrollIntoView({behavior: 'smooth', block: 'center'});
+      });
 
-    placeMarker.addListener('mouseover', function () {
-      placeMarker.setIcon(SELECTED_PIN_PATH);
-    });
+      placeMarker.addListener('mouseover', function () {
+        placeMarker.setIcon(SELECTED_PIN_PATH);
+      });
 
-    placeMarker.addListener('mouseout', function () {
-      if (placeMarker != focusedPin) {
-        placeMarker.setIcon(PIN_PATH);
-      }
-    });
+      placeMarker.addListener('mouseout', function () {
+        if (placeMarker != focusedPin) {
+          placeMarker.setIcon(PIN_PATH);
+        }
+      });
 
-    markers.push(placeMarker);
+      markers.push(placeMarker);
+    }
   }
 
   document.getElementById(SCROLL_ID).hidden = false;
-
   // Handle favoriting a place
   $('.icon').click(function() {
+    const name = $(this).parent().parent().parent().attr('placeName');
+    const placeId = $(this).parent().next().next().next().attr('savedPlaceId');
     $(this).toggleClass('press');
     if (firebase.auth().currentUser && $(this).hasClass('press')) {
-      const name = $(this).parent().parent().parent().attr('placeName');
-      const link = $(this).parent().next().attr('href');
       const time = $(this).parent().next().next().text();
        // Add users saved places to the real time database in Firebase when star is pressed
       const database = firebase.database();
       var uID = firebase.auth().currentUser.uid;
-      var ref = database.ref('users/' + uID + '/' + 'places' + '/' + name);
+      var ref = database.ref('users/' + uID + '/places/' + name);
       var data = {
         name: name,
-        link: link,
-        time: time,
+        timeAsString: time,
+        place_id: placeId,
       }
       ref.set(data);
+      addGeometry(name, placeId);
```

Comment:
Just curious, is it possible to add the latitude and longitude as hidden attributes in the similar to savedPlaceId on line 541? Since they're float values I imagine it may difficult or not even possible, but if so, it could be worth looking into to avoid additional API calls.

Reply:
done.  That was a good idea.